metrics-jetty10/src/main/java/io/dropwizard/metrics/jetty10/InstrumentedHandler.java

```diff
@@ -0,0 +1,379 @@
+    private static final String NAME_HEAD_REQUESTS = "head-requests";
+    private static final String NAME_PUT_REQUESTS = "put-requests";
+    private static final String NAME_DELETE_REQUESTS = "delete-requests";
+    private static final String NAME_OPTIONS_REQUESTS = "options-requests";
+    private static final String NAME_TRACE_REQUESTS = "trace-requests";
+    private static final String NAME_CONNECT_REQUESTS = "connect-requests";
+    private static final String NAME_MOVE_REQUESTS = "move-requests";
+    private static final String NAME_OTHER_REQUESTS = "other-requests";
+    private static final String NAME_PERCENT_4XX_1M = "percent-4xx-1m";
+    private static final String NAME_PERCENT_4XX_5M = "percent-4xx-5m";
+    private static final String NAME_PERCENT_4XX_15M = "percent-4xx-15m";
+    private static final String NAME_PERCENT_5XX_1M = "percent-5xx-1m";
+    private static final String NAME_PERCENT_5XX_5M = "percent-5xx-5m";
+    private static final String NAME_PERCENT_5XX_15M = "percent-5xx-15m";
+
+    private final MetricRegistry metricRegistry;
+
+    private String name;
+    private final String prefix;
+
+    // the requests handled by this handler, excluding active
+    private Timer requests;
+
+    // the number of dispatches seen by this handler, excluding active
+    private Timer dispatches;
+
+    // the number of active requests
+    private Counter activeRequests;
+
+    // the number of active dispatches
+    private Counter activeDispatches;
+
+    // the number of requests currently suspended.
+    private Counter activeSuspended;
+
+    // the number of requests that have been asynchronously dispatched
+    private Meter asyncDispatches;
+
+    // the number of requests that expired while suspended
+    private Meter asyncTimeouts;
+
+    private Meter[] responses;
+
+    private Timer getRequests;
+    private Timer postRequests;
+    private Timer headRequests;
+    private Timer putRequests;
+    private Timer deleteRequests;
+    private Timer optionsRequests;
+    private Timer traceRequests;
+    private Timer connectRequests;
+    private Timer moveRequests;
+    private Timer otherRequests;
+
+    private AsyncListener listener;
+
+    private HttpChannelState.State DISPATCHED_HACK;
+
+    /**
+     * Create a new instrumented handler using a given metrics registry.
+     *
+     * @param registry the registry for the metrics
+     */
+    public InstrumentedHandler(MetricRegistry registry) {
+        this(registry, null);
+    }
+
+    /**
+     * Create a new instrumented handler using a given metrics registry.
+     *
+     * @param registry the registry for the metrics
+     * @param prefix   the prefix to use for the metrics names
+     */
+    public InstrumentedHandler(MetricRegistry registry, String prefix) {
+        this.metricRegistry = registry;
+        this.prefix = prefix;
+
+        try {
+            DISPATCHED_HACK = HttpChannelState.State.valueOf("HANDLING");
```

Comment:
I believe we don't need this hack in Jetty 10/11 as `HttpChannelState.State.HANDLING` will always be present there.

Reply:
@arteam Thank, you're absolutely right!